src/chat/KeywordResponses.hpp

```diff
@@ -0,0 +1,38 @@
+// chris busch
+
+#pragma once
+
+namespace chat {
+
+/**
+ * A response message with an assigned weight that's used when
+ * determining a random response to a keyword from the user.
+ */
+struct WeightedResponse {
+    int weight_;
+    char *message_;
+};
+
+/**
+ * Represents an keyword rule that is matched against the message supplied
+ * by the player, and the collection of response messages that can be sent
+ * if a keyword match is found.  Each keyword rule can contain logical operators.
+ * e.g. (what is your name|who are you).
+ * Each instance of KeywordResponses belongs to a specific Database.
+ */
+class KeywordResponses {
+public:
+    int add_keywords(char *);
+    char *get_keywords() { return keywords_; }
```

Comment:
Some of these functions can be made member-const:

```cpp
char *get_keywords() const { return keywords_; }
const WeightedResponse &get_response(int num) const ... // IFF we can return a non-mutable
const char *get_random_response() const;
```

Const-correctness is another C++ superpower. It's not perfect, but starting with it is easier than retrofitting.

Ideally all those can return `const`  too. In some cases both are needed, which can be useful sometimes (this is how e.g. copy on write can be implemented in C++, you know when a caller needs a mutable reference and clone yourself etc).